Assets/ZSSRichTextEditor.js

```diff
@@ -273,49 +273,66 @@ ZSSEditor.getJoinedCaretArguments = function() {
     return joinedArguments;
 };
 
-ZSSEditor.getYCaretInfo = function() {
-    var y = 0, height = 0;
+ZSSEditor.getCaretYPosition = function() {
     var sel = window.getSelection();
-    if (sel.rangeCount) {
+    // Next line is comented to prevent deselecting selection. It looks like work but if there are any issues will appear then uconmment it as well as code above.
+    //sel.collapseToStart();
+    var range = sel.getRangeAt(0);
+    var span = document.createElement('span');// something happening here preventing selection of elements
+    range.insertNode(span);
+    var topPosition = span.offsetTop;
+    span.parentNode.removeChild(span);
+    return topPosition;
+}
+
+ZSSEditor.getYCaretInfo = function() {
+    var y = 0;
+    var height = 0;
+    var selection = window.getSelection();
+    
+    if (! selection.rangeCount) { // if no selection available just return 0
+        this.caretInfo.y = y;
+        this.caretInfo.height = height;
+        return this.caretInfo;
+    }
+    
+    var range = selection.getRangeAt(0);
+    var needsToWorkAroundNewlineBug = (range.startOffset == 0 || range.getClientRects().length == 0);
```

Comment:
We can remove this variable completely, since `range.startOffset == 0` was being used to detect problems with the `y` position calculation.

The only condition we need to check is `range.getClientRects().length == 0`, but we're already doing it far below... search my comment below for more info.